src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -247,56 +342,74 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
         /// <param name="callbackObject">Asynchronous I/O callback object</param>
         /// <param name="parallel">Should MultiSubnetFailover be used</param>
         /// <returns>SNITCPHandle</returns>
-        private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, object callbackObject, bool parallel)
+        private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, object callbackObject, bool parallel, ref byte[] spnBuffer)
         {
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
 
-            if (string.IsNullOrWhiteSpace(fullServerName))
+            string fqdnHostName = null; ;
+            int port = -1;
+            Exception exception = null;
+
+            if (string.IsNullOrWhiteSpace(fullServerName)) // when fullServerName is empty
             {
-                int defaultInstancePort = TryToGetDefaultInstancePort(DefaultHostname);
-                return new SNITCPHandle(DefaultHostname, (defaultInstancePort > 0 ? defaultInstancePort : DefaultSqlServerPort), timerExpire, callbackObject, parallel);
+                fqdnHostName = GetFullyQualifiedDomainName(DefaultHostName);
+                port = DefaultSqlServerPort;
+                spnBuffer = GetMsSqlServerSPN(fqdnHostName, port);
+                return new SNITCPHandle(fqdnHostName, port, timerExpire, callbackObject, parallel);
             }
 
             string[] serverNamePartsByComma = fullServerName.Split(CommaSeparator);
             string[] serverNamePartsByBackSlash = fullServerName.Split(BackSlashSeparator);
 
+            // when no port or instance name provided
             if (serverNamePartsByComma.Length < 2 && serverNamePartsByBackSlash.Length < 2)
             {
-                return new SNITCPHandle(fullServerName, DefaultSqlServerPort, timerExpire, callbackObject, parallel);
+                fqdnHostName = GetFullyQualifiedDomainName(fullServerName);
+                port = DefaultSqlServerPort;
             }
+            // when port is provided, and no instance name
             else if (serverNamePartsByComma.Length == 2 && serverNamePartsByBackSlash.Length < 2)
             {
-                string hostName = serverNamePartsByComma[0];
-                int portNumber = -1;
+                fqdnHostName = GetFullyQualifiedDomainName(serverNamePartsByComma[0]);
```

Comment:
The current Tcp connectivity behaviour should be left as - is and no reverse lookup is needed to connect to Sql Server. This is going to add a network round trip for every connection. 
The SPN is needed only when we request a Kerberos ticket.